src/hooks/useRecentSearches.ts

```diff
@@ -1,41 +1,49 @@
 import { useCallback, useEffect, useRef, useState } from 'react';
 import RecentSearches, { ISearch } from 'recent-searches';
 
-export const RECENT_SEARCHES_KEY = '__yxt_recent_searches__';
-
 export function useRecentSearches(
-  recentSearchesLimit: number
-): [ISearch[]|undefined, (input: string) => void, () => void] {
+  recentSearchesLimit: number,
+  verticalKey: string | null
+): [ISearch[] | undefined, (input: string) => void, () => void] {
   const recentSearchesLimitRef = useRef(recentSearchesLimit);
-  const [ recentSearches, setRecentSeaches ] = useState<RecentSearches>(
+  const [recentSearchesKey, setRecentSearchesKey] = useState(getRecentSearchesKey(verticalKey));
+  const [recentSearches, setRecentSeaches] = useState<RecentSearches>(
     new RecentSearches({
```

Comment:
I feel maybe like this RecentSearches instance should be in a ref instead of in state?